tools/bpf_inspect.py

```diff
@@ -86,13 +85,53 @@ def get_tramp_progs(bpf_prog):
                 yield tramp_aux.prog
 
 
+def get_legacy_tramp_progs(bpf_prog):
+    try:
+        tr = bpf_prog.aux.member_("trampoline")
+    except LookupError:
+        # trampoline is available since Linux kernel commit
+        # fec56f5890d9 ("bpf: Introduce BPF trampoline") (in v5.5).
+        # Skip trampoline if current kernel doesn't support it.
+        return []
+
+    return get_prog_from_tramp(tr)
+
+
+def get_tramp_progs(bpf_prog):
+    try:
+        tramp = bpf_prog.aux.member_("dst_trampoline")
+    except LookupError:
+        # dst_rampoline replaces "trampoline" since Linux kernel commit
+        # 3aac1ead5eb6 ("bpf: Move prog->aux->linked_prog and trampoline into
+        # bpf_link on attach") (in v5.10).
+        # Skip if current kernel doesn't support it.
+        return []
```

Comment:
Can we have `get_tramp_progs()` handle all possible kernels instead of having a separate `get_legacy_tramp_progs()`? Something like:

```suggestion
    # "dst_trampoline" replaces "trampoline" since Linux kernel commit
    # 3aac1ead5eb6 ("bpf: Move prog->aux->linked_prog and trampoline into
    # bpf_link on attach") (in v5.10).
    try:
        tramp = bpf_prog.aux.member_("dst_trampoline")
    except LookupError:
        # Before that, "trampoline" is available since Linux kernel commit
        # fec56f5890d9 ("bpf: Introduce BPF trampoline") (in v5.5).
        try:
            tramp = bpf_prog.aux.member_("trampoline")
        except LookupError:
            # This kernel doesn't support trampolines at all.
            return
        yield from get_prog_from_tramp(tramp)
        return
```